radicle-httpd/src/api/test.rs

```diff
@@ -0,0 +1,128 @@
+
+pub fn seed() -> Context {
+    let random_suffix = Alphanumeric.sample_string(&mut rand::thread_rng(), 16);
+    let temp_dir = env::temp_dir()
+        .join("radicle-httpd-tests")
+        .join(random_suffix);
+    let workdir = temp_dir.join("hello-world");
+    let rad_home = temp_dir.join("radicle");
+
+    env::set_var("RAD_PASSPHRASE", "asdf");
+    env::set_var("RAD_DEBUG", "1");
+
+    fs::create_dir_all(&workdir).unwrap();
+    fs::create_dir_all(&rad_home).unwrap();
+
+    // add commits to WORKDIR
+    let repo = git2::Repository::init(&workdir).unwrap();
+    let tree =
+        radicle::git::write_tree(Path::new("README"), "Hello World!\n".as_bytes(), &repo).unwrap();
+
+    let sig_time = git2::Time::new(1673001014, 0);
+    let sig = git2::Signature::new("Alice Liddell", "alice@radicle.xyz", &sig_time).unwrap();
+
+    let oid = repo
+        .commit(Some("HEAD"), &sig, &sig, "Initial commit\n", &tree, &[])
+        .unwrap();
+    let commit = repo.find_commit(oid).unwrap();
+
+    repo.checkout_tree(tree.as_object(), None).unwrap();
+
+    fs::create_dir(workdir.join("dir1")).unwrap();
+    fs::write(
+        workdir.join("dir1").join("README"),
+        "Hello World from dir1!\n",
+    )
+    .unwrap();
+    let mut index = repo.index().unwrap();
+    index
+        .add_all(["."], git2::IndexAddOption::DEFAULT, None)
+        .unwrap();
+    index.write().unwrap();
+
+    let oid = index.write_tree().unwrap();
+    let tree = repo.find_tree(oid).unwrap();
+
+    repo.commit(
+        Some("HEAD"),
+        &sig,
+        &sig,
+        "Add another folder\n",
+        &tree,
+        &[&commit],
+    )
+    .unwrap();
+
+    // eq. rad auth
+    let profile = radicle::Profile::init(rad_home, "asdf".to_owned()).unwrap();
+
+    // rad init
+    rad_init::init(
+        rad_init::Options {
+            path: Some(workdir.clone()),
+            name: Some("hello-world".to_string()),
+            description: Some("Rad repository for tests".to_string()),
+            branch: None,
+            interactive: false.into(),
+            setup_signing: false,
+            set_upstream: false,
+        },
+        &profile,
+    )
+    .unwrap();
+
+    // eq. rad issue new
+    env::set_var("RAD_COMMIT_TIME", "1673001014");
+
+    let signer = radicle_cli::terminal::signer(&profile).unwrap();
+    let storage = &profile.storage;
+    let (_, id) = radicle::rad::repo(&workdir).unwrap();
+    let repo = storage.repository(id).unwrap();
+    let mut issues = Issues::open(*signer.public_key(), &repo).unwrap();
+    issues
+        .create(
+            "Issue #1".to_string(),
+            "Change 'hello world' to 'hello everyone'".to_string(),
+            &[],
+            &signer,
+        )
+        .unwrap();
+
+    Context {
+        profile: Arc::new(profile),
+        sessions: Default::default(),
+    }
+}
+
+pub async fn request(app: &Router, path: String) -> Response {
```

Comment:
`request` can be re-written to take `path: impl ToString`, and then we can do:

```
let response = request(&app, "/stats").await;
```



instead of 
```
let response = request(&app, "/stats".to_string()).await;
```

but at the cost of doing another allocation inside the `request`:
```
Request::builder().uri(path) ---> Request::builder().uri(path.to_string())
```

worth it?

Reply:
Yeah worth it!

Reply:
https://github.com/radicle-dev/heartwood/compare/cf86b516b8ef8e58885a0c38b7c015113a3dd5cf..9114dccc40aa5a5e5f244b24184eea1d21b21cd3